src/storage/src/client.rs

```diff
@@ -0,0 +1,325 @@
+///   authentication may need to override this default.
+///
+/// # Pooling and Cloning
+///
+/// `Storage` holds a connection pool internally, it is advised to
+/// create one and the reuse it.  You do not need to wrap `Storage` in
+/// an [Rc](std::rc::Rc) or [Arc](std::sync::Arc) to reuse it, because it
+/// already uses an `Arc` internally.
+///
+/// # Service Description
+///
+/// The Cloud Storage API allows applications to read and write data through
+/// the abstractions of buckets and objects. For a description of these
+/// abstractions please see <https://cloud.google.com/storage/docs>.
+///
+/// This client is used to perform metadata operations, such as creating
+/// buckets, deleting objects, listing objects, etc. It does not expose any
+/// functions to write or read data in objects.
+///
+/// Resources are named as follows:
+///
+/// - Projects are referred to as they are defined by the Resource Manager API,
+///   using strings like `projects/123456` or `projects/my-string-id`.
+///
+/// - Buckets are named using string names of the form:
+///   `projects/{project}/buckets/{bucket}`
+///   For globally unique buckets, `_` may be substituted for the project.
+///
+/// - Objects are uniquely identified by their name along with the name of the
+///   bucket they belong to, as separate strings in this API. For example:
+///   ```no_rust
+///   bucket = "projects/_/buckets/my-bucket"
+///   object = "my-object/with/a/folder-like/name"
+///   ```
+///   Note that object names can contain `/` characters, which are treated as
+///   any other character (no special directory semantics).
+///
+/// [with_endpoint()]: ClientBuilder::with_endpoint
+/// [with_credentials()]: ClientBuilder::with_credentials
+/// [Private Google Access with VPC Service Controls]: https://cloud.google.com/vpc-service-controls/docs/private-connectivity
+/// [Application Default Credentials]: https://cloud.google.com/docs/authentication#adc
+#[derive(Clone, Debug)]
+pub struct Storage {
+    inner: reqwest::Client,
+    cred: auth::credentials::Credentials,
+    endpoint: String,
+}
+
+impl Storage {
+    /// Returns a builder for [Storage].
+    ///
+    /// ```no_run
+    /// # tokio_test::block_on(async {
+    /// # use google_cloud_storage::client::Storage;
+    /// let client = Storage::builder().build().await?;
+    /// # gax::Result::<()>::Ok(()) });
+    /// ```
+    pub fn builder() -> ClientBuilder {
+        gax::client_builder::internal::new_builder(client_builder::Factory)
+    }
+
+    /// A simple upload from a buffer.
+    ///
+    /// # Example
+    /// ```
+    /// # use google_cloud_storage::client::Storage;
+    /// async fn example(client: &Storage) -> gax::Result<()> {
+    ///     let response = client
+    ///         .insert_object(
+    ///             "projects/_/buckets/my-bucket",
+    ///             "my-object",
+    ///             bytes::Bytes::from("the quick brown fox jumped over the lazy dog"),
```

Comment:
nit: `bytes::Bytes::from()` is unnecessary, but maybe you are just showing that it is not necessarily a string

Reply:
Fixed.